robot/robot.go

```diff
@@ -127,23 +127,6 @@ type Status struct {
 	Status interface{}
 }
 
-// AllResourcesByName returns an array of all resources that have this simple name.
-func AllResourcesByName(r Robot, name string) []resource.Resource {
```

Comment:
Why are we removing this? This is likely to break a number of people's scripts.

Reply:
This is no longer being used anywhere in the codebase

Reply:
@cheukt, thoughts on this?

Reply:
I would vote for deprecating this method (use a `Deprecate` comment) and not removing it. It is encouraging users of the Go SDK to refer to resources by short name only and deal with potential duplicates on their own, which is not great. But @biotinker is right that it's dangerous to remove public API post-GA. We can remove it when we hit the next major version.

Reply:
agree with @biotinker  in that I don't think we can remove it at this point and users may find it useful for other things. I'm not sure we need to deprecate it right now either, but would be in favor of adding a comment that this is not recommended and may become deprecated later

Reply:
Added the comment and restored the function